collect_app/src/main/java/org/odk/collect/android/entities/JsonFileEntitiesRepository.kt

```diff
@@ -81,20 +93,26 @@ class JsonFileEntitiesRepository(directory: File) : EntitiesRepository {
             entitiesFile.createNewFile()
         }
 
-        val typeToken = TypeToken.getParameterized(
-            MutableMap::class.java,
-            String::class.java,
-            TypeToken.getParameterized(MutableList::class.java, JsonEntity::class.java).type
-        )
+        try {
+            val typeToken = TypeToken.getParameterized(
+                MutableMap::class.java,
+                String::class.java,
+                TypeToken.getParameterized(MutableList::class.java, JsonEntity::class.java).type
+            )
 
-        val json = entitiesFile.readText()
-        return if (json.isNotBlank()) {
-            val parsedJson =
-                Gson().fromJson<MutableMap<String, MutableList<JsonEntity>>>(json, typeToken.type)
+            val json = entitiesFile.readText()
+            return if (json.isNotBlank()) {
+                val parsedJson =
+                    Gson().fromJson<MutableMap<String, MutableList<JsonEntity>>>(json, typeToken.type)
 
-            parsedJson
-        } else {
-            mutableMapOf()
+                parsedJson
+            } else {
+                mutableMapOf()
+            }
+        } catch (e: Exception) {
+            entitiesFile.delete()
+            entitiesFile.createNewFile()
```

Comment:
If I comment those two lines out this test still passes: https://github.com/getodk/collect/pull/6132/files#diff-f7e8798917d3ccfe8e394d1d714eb9535648a9c63a42c41b5ec30d32006cbd2cR35